en/dev_log/ulog_file_format.md

```diff
@@ -41,233 +47,329 @@ The file consists of three sections:
+struct message_header_s {
+  uint16_t msg_size;
+  uint8_t msg_type;
+};
+```
+
+- `msg_size` is the size of the message in bytes without the header.
+- `msg_type` defines the content, and is a single character.
+
+:::note
+Message sections below are prefixed with the character that corresponds to it's `msg_type`.
+:::
 
 
 ### Definitions Section
 
-Variable length section, contains version information, format definitions, and (initial) parameter values.
+The definitions section contains basic information such as software version, message format, initial parameter values, and so on.
+
+The message types in this section are:
+1. [Flag Bits](#b-flag-bits-message)
+2. [Format Definition](#f-format-message)
+3. [Information](#i-information-message)
+4. [Multi Information](#m-multi-information-message)
+5. [Parameter](#p-parameter-message)
+6. [Default Parameter](#q-default-parameter-message)
+
+
+#### 'B': Flag Bits Message
+
+:::note
+This message must be the **first message** right after the header section, so that it has a fixed constant offset from the start of the file!
+:::
+
+This message provides information to the log parser whether the log is parsable or not.
 
-The Definitions and Data sections consist of a stream of messages.
-Each starts with this header:
 ```c
-struct message_header_s {
-	uint16_t msg_size;
-	uint8_t msg_type
+struct ulog_message_flag_bits_s {
+  struct message_header_s header; // msg_type = 'B'
+  uint8_t compat_flags[8];
+  uint8_t incompat_flags[8];
+  uint64_t appended_offsets[3]; // file offset(s) for appended data if appending bit is set
 };
 ```
 
-`msg_size` is the size of the message in bytes without the header (`hdr_size`= 3 bytes).
-`msg_type` defines the content and is one of the following:
+- `compat_flags`: compatible flag bits
+  - These flags indicate the presence of features in the log file that are compatible with any ULog parser.
+  - `compat_flags[0]`: *DEFAULT_PARAMETERS* (Bit 0): if set, the log contains [default parameters message](#q-default-parameter-message)
+
+  The rest of the bits are currently not defined and must be set to 0.
+  These bits can be used for future ULog changes that are compatible with existing parsers.
+  For example, adding a new message type can be indicated by defining a new bit in the standard, and existing parsers will ignore the new message type.
+  It means parsers can just ignore the bits if one of the unknown bits is set.
+- `incompat_flags`: incompatible flag bits.
+  - `incompat_flags[0]`: *DATA_APPENDED* (Bit 0): if set, the log contains appended data and at least one of the `appended_offsets` is non-zero.
+  
+  The rest of the bits are currently not defined and must be set to 0.
+  This can be used to introduce breaking changes that existing parsers cannot handle. For example, when an old ULog Parser that didn't have the concept of *DATA_APPENDED* reads the newer ULog, it would stop parsing the log as the log will contain out-of-spec messages / concepts.
```

Comment:
```suggestion
  This can be used to introduce breaking changes that existing parsers cannot handle. For example, when an old ULog parser that didn't have the concept of *DATA_APPENDED* reads the newer ULog, it would stop parsing the log as the log will contain out-of-spec messages / concepts.
```